app/interactors/deployers/heroku.rb

```diff
@@ -0,0 +1,85 @@
+require "platform-api"
+
+module Deployers
+  class Heroku
+    attr_reader :project, :environment_name
+
+    class Logger
+      attr_reader :streams
+
+      def initialize(*streams)
+        @streams = streams
+      end
+
+      def output(str)
+        streams.each { |stream| stream.puts str }
+      end
+      alias info output
+      alias debug output
+    end
+
+    def initialize(project, environment_name)
+      @project = project
+      @environment_name = environment_name
+    end
+
+    def last_deploy_commit
+      releases = heroku.release.list(app_name).to_a
+      id = releases.last.fetch("slug").fetch("id")
+      slug = heroku.slug.info(app_name, id)
+      slug.fetch("commit")
+    end
+
+    def deploy(deploy, options={})
+      begin
+        project.repo.refresh!
+
+        rugged_repo = project.repo.send(:connection)
+        branch_ref = rugged_repo.refs.find { |ref| ref.target_id == deploy.sha }
+        heroku_remote = rugged_repo.remotes.create_anonymous(heroku_git_url)
+
+        # The + in the refspec means --force
```

Comment:
Interesting!

Reply:
I know! It took a while of me searching "How in the world do you force with `libgit2`?" 😅 And of course, someone on SO was "it's documented in git's refspec, so I don't know why you're complaining it's not documented". 🙃